### PR TITLE
[NT-1269, NT-1270] Deprecate BackingViewController

### PR DESCRIPTION
**Pull request by justinswart**
# 📲 What

- Deprecates all old references to `BackingViewController`, which is now replaced by `ManagePledgeViewController`.
- Allows a creator to view a backing from a message thread via the dashboard.

# 🤔 Why

`BackingViewController` was still left over from the work that we did to re-build our checkout flow natively. This PR removes it and updates the callsites that configured and presented it.

# 🛠 How

- Refactored `ManagePledgeViewController` to support only being configured by a Project param (slug or id) and a Backing param (id).
  - `ManagePledgeViewModel` will now always fetch the relevant `Project` and `Backing` instead of only sometimes fetching the `Project` as before. This just helped to simplify this view model as it was becoming unwieldy trying to handle all of the configuration scenarios.
- Updated callsites and configuration in the following view controllers:
  - `ActivitiesViewController`.
  - `MessagesViewController`.
  - `ProjectActivitiesViewController`.
  - `ProjectPamphletContentViewController`.
  - `ProjectPamphletViewController`.
- Updated `MessagesViewModel` to retrieve the backing ID from the `MessageThread`.
  - **Note:** I wasn't able to find a scenario in which the `MessageThread` did _not_ have a `Backing` (unless the thread was between a creator and a user that hadn't backed of course). I would just want to be sure there isn't a use case that I'm unaware of in which the thread might not have a `Backing`.

# 👀 See

| Before 🐛 | After 🦋 |
| --- | --- |
| image | image |

# ✅ Acceptance criteria

- [ ] Test navigating to manage/view a backing from `ActivitiesViewController`.
- [ ] Test navigating to manage/view a backing from `MessagesViewController`.
  - This is accessible via the dashboard logged in as a creator that has a message thread with a backer.
- [ ] Test navigating to manage/view a backing from `ProjectActivitiesViewController`.
- [ ] Test navigating to manage/view a backing from `ProjectPamphletContentViewController`.
- [ ] Test navigating to manage/view a backing from `ProjectPamphletViewController`.

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Library/SharedFunctions.swift [hunk 1]
```diff
@@ -64,6 +64,23 @@ internal func reward(from backing: Backing, inProject project: Project) -> Rewar
...
+/**
+ Returns a reward for a backing ID in a given project
+
+ - parameter backing: A backing ID
+ - parameter project: A project
+
+ - returns: A reward
+ */
+
+internal func reward(withId rewardId: Int, inProject project: Project) -> Reward {
```
Comment (ifbarrera) [inline Library/SharedFunctions.swift:76]: We also have this function in `SharedFunctions.swift`:
```
/**
 Returns a reward from a backing in a given project

 - parameter backing: A backing
 - parameter project: A project

 - returns: A reward
 */

internal func reward(from backing: Backing, inProject project: Project) -> Reward {
  let noRewardFromProject = project.rewards.first { $0.id == Reward.noReward.id }

  return backing.reward
    ?? project.rewards.first { $0.id == backing.rewardId }
    ?? noRewardFromProject
    ?? Reward.noReward
}
```
Seems pretty much identical, is it worthwhile to consolidate so we don't have to maintain this logic in two places?

File: Library/ViewModels/ManagePledgeViewModel.swift [hunk 2]
```diff
@@ -121,17 +134,23 @@ public final class ManagePledgeViewModel:
     )
     .ignoreValues()
 
-    let projectAndReward = project
-      .filterMap { project in
-        guard let backing = project.personalization.backing else {
-          return nil
-        }
+    let userIsCreatorOfProject = project.map {
+      project in project.creator.id == AppEnvironment.current.currentUser?.id
```
Comment (ifbarrera) [inline Library/ViewModels/ManagePledgeViewModel.swift:138]: You could use the shared function `func currentUserIsCreator(of project: Project)` here 😅 

It's in... you guess it, `SharedFunctions.swift`

File: Library/ViewModels/MessagesViewModel.swift [hunk 3]
```diff
@@ -163,11 +163,16 @@ public final class MessagesViewModel: MessagesViewModelType, MessagesViewModelIn
 
     self.goToBacking = Signal.combineLatest(messageThreadEnvelope, currentUser)
       .takeWhen(self.backingInfoPressedProperty.signal)
-      .map { env, currentUser in
-        env.messageThread.project.personalization.isBacking == .some(true)
-          ? (env.messageThread.project, currentUser)
-          : (env.messageThread.project, env.messageThread.participant)
+      .map { env, _ -> ManagePledgeViewParamConfigData? in
```
Comment (ifbarrera) [inline Library/ViewModels/MessagesViewModel.swift:166]: Could also do `filterMap` here so you don't need to add the `skipsNil()`

File: Library/ViewModels/ManagePledgeViewModel.swift [hunk 4]
```diff
@@ -69,32 +77,49 @@ public final class ManagePledgeViewModel:
...
+
+    let shouldFetchGraphBackingWithParam = Signal.merge(
+      backingParam,
+      backingParam.takeWhen(shouldBeginRefresh)
     )
 
-    let graphBackingEvent = shouldFetchGraphBackingWithProject
-      .switchMap { project in
+    let graphBackingEvent = shouldFetchGraphBackingWithParam
+      .logEvents(identifier: "***")
```
Comment (ifbarrera) [inline Library/ViewModels/ManagePledgeViewModel.swift:116]: Log here 😄

**Pull request merged**